x/qgb/keeper/query_valset.go

```diff
@@ -41,3 +42,22 @@ func (k Keeper) ValsetRequestByNonce(
 
 	return nil, sdkerrors.Wrap(sdkerrors.ErrNotFound, "valset request nonce not found")
 }
+
+// LastValsetBeforeHeight queries the last valset request before height
+func (k Keeper) LastValsetBeforeHeight(
+	c context.Context,
+	req *types.QueryLastValsetBeforeHeightRequest) (*types.QueryLastValsetBeforeHeightResponse, error) {
+	valReq := k.GetValsets(sdk.UnwrapSDKContext(c))
+	for _, valset := range valReq {
+		if !k.HasValsetRequest(sdk.UnwrapSDKContext(c), valset.Nonce+1) ||
```

Comment:
What if there's only a single valset saved, with height >= `req.Height`? Then this condition would incorrectly be `true`.

Reply:
AFAIU, it can't be. We will always have a valset at block 0 or 1

Reply:
This is the condition creating valsets:
```
(latestValset == nil) || (lastUnbondingHeight == uint64(ctx.BlockHeight())) || significantPowerDiff
```
A valset will be created:
- If we have no valset 
- We're an unbonding height
- There  was a significant power difference in the validator set

Reply:
Please document this invariant here then in a comment.

Reply:
Please check now if the comment is good.